daemon/call.c

```diff
@@ -2787,6 +2787,13 @@ static void __media_init_from_flags(struct call_media *other_media, struct call_
 		if (sp->desired_family)
 			media->desired_family = sp->desired_family;
 	}
+
+	if (flags->opmode == OP_OFFER) {
+		other_media->media_rec_slot = flags->media_rec_slot_offer;
+		media->media_rec_slot = flags->media_rec_slot_answer;
+		media->media_rec_slots = flags->media_rec_slots;
```

Comment:
I assume the `media_rec_slots` value should be the same for all parties to a call within the same call? Perhaps store the value in the `call` instead then, and make sure it's set only once? It might make the logic in `mix_get_index` easier as well. Or you could even have it as a global config variable (in `rtpengine` or in `rtpengine-recording`)

Reply:
I wasn't keen on the global for `media_rec_slots`, in case the rtpengine instance was having to deal with multiple call scenarios (SIPREC with > 2 media, regular call with offer/answer etc). 

Yes it would be set to the same for all parties to a call within the same call. I'll look at storing it in the `call`. Could you point me to a sensible place to do this?

Reply:
> I wasn't keen on the global for `media_rec_slots`, in case the rtpengine instance was having to deal with multiple call scenarios (SIPREC with > 2 media, regular call with offer/answer etc).
> 
> Yes it would be set to the same for all parties to a call within the same call. I'll look at storing it in the `call`. Could you point me to a sensible place to do this?

You can just put the values into `struct call` and leave the assignment in the same place. There isn't (yet) a dedicated function to assign `call` values from `flags`, but you could (ab)use `__call_monologue_init_from_flags()` and just access it through `ml->call`. There are some other call-specific flags assigned there already.

Reply:
@rfuchs  I think I am almost here with this. I am struggling to find/assign the "slot" from the offer/answer whenever a play_media happens. I can see that `call_play_media_ng` is called and that `setup_stream_proc` then gets called again to create some extra streams in the recording metafile. At this point I need to find the "slot" that is already set from the corresponding `media` or `monologue` but I can't see where this has to happen. Do you have any suggestions?

Reply:
@rfuchs please ignore the above, it was a bug in an OpenSIPS configuration when calling rtpengine...